### DISCUSSION TITLE
Setting up nanobind, stubs, sphinx and automodule

### DISCUSSION THREAD

**Discussion by JeanDidierGaraud** [wjakob/nanobind#824] [Q&A]
Hello,
I'm using sphinx, automodule and autosummary for my package's documentation (and nanobind, obviously 😉)

I forked nanobind_example to make a MRE [1].

With basic configuration (i.e. no docstring signature, no nanobind_stubs), only autofunction does a pretty rendering of functions' annotations (this was already discussed in #420).

Adding a `nanobind_example_ext.pyi` stub does not improve it (maybe it's related to this [currently open sphinx issue](https://github.com/sphinx-doc/sphinx/issues/7630)?).

If I rename the stub as .py and import it (see this commit [2]), everything is nicely formatted as expected.

Is it the correct way to go?

[1] https://github.com/JeanDidierGaraud/nanobind_example
[2] https://github.com/JeanDidierGaraud/nanobind_example/commit/34dee8c90c6372aa5d6bfaf2300ce182d5ebe182

JD

**Comment by wojdyr** [comment]
Shouldn't the stub be in `nanobind_example/__init__.pyi` instead of `nanobind_example_ext.pyi`?
(just a thought, no idea what works with sphinx)

**Reply by JeanDidierGaraud** [reply]
I gave it a try, it also seems to be ignored by sphinx.
